### PR TITLE
feat: prisma

### PR DESCRIPTION
**Pull request by JuanPablo2655**
**Please describe the changes this PR makes and why it should be merged:**
Making the switch to Postgres and Prisma

**Status and versioning classification:**
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- This PR includes new features.

### CODE REVIEW

**Comment by JuanPablo2655** [comment]
bruh, I didn't think [Volta](https://volta.net/) would mess with GitHub.

File: src/commands/utility/checksocialcredit.ts [hunk 1]
```diff
@@ -17,15 +18,16 @@ abstract class checkSocialCredit extends GenericCommand {
 	}
 
 	async run(message: Message, args: string[], addCD: () => Promise<void>) {
-		const member = await resolveMember(args[0], message.guild!).catch(() => message.member);
+		if (!message.guild) throw new UserError({ identifier: Identifiers.PreconditionGuildOnly }, 'guild only command');
+		const member = await resolveMember(args[0], message.guild).catch(() => message.member);
+		if (!member) throw new Error('missing member');
+		const user = (await getUser(member.id)) ?? (await createUser(member.id));
+		if (!user) throw new Error('missing user database entry');
```
Comment (rand0m-cloud) [inline src/commands/utility/checksocialcredit.ts:25]: this is a good example of code that could be refactored into methods. `getOrCreate{..}` and `get{..}` would be cleaner methods and I would place the error throwing in those methods.

File: src/lib/database/bot.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+		return await db.bot.create({
+			data: {
+				id: id,
+				...data,
+			},
+		});
+	} catch (err) {
+		const error = err as Error;
+		logger.error(error, error.message);
+	}
```
Comment (rand0m-cloud) [inline src/lib/database/bot.ts:22]: this is a nice try catch block pattern that could be abstracted out for reuse. similar to the `.unwrap_or_log()` you see in rust with logging.

**Review by rand0m-cloud — APPROVED** [review]
LGTM, there are some opportunities for refactoring out common patterns but that is not mission critical.

**Pull request merged**